src/Tickets/Commerce/Promoter_Observer.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace TEC\Tickets\Commerce;
+
+use Tribe\Tickets\Promoter\Triggers\Contracts\Attendee_Model;
+use Tribe\Tickets\Promoter\Triggers\Models\Attendee;
+
+class Promoter_Observer {
+
+	/**
+	 * @since 4.12.0
+	 *
+	 * @var Tribe__Tickets__Promoter__Observer $observer ET Observer reference.
+	 */
+	private $observer;
+
+	/**
+	 * Constructor.
+	 *
+	 */
+	public function __construct() {
+		$this->observer = tribe( \Tribe__Tickets__Promoter__Observer::class );
+		$this->hook();
+	}
+
+	/**
+	 * Attach hooks for trigger messages.
+	 *
+	 * @since TBD
+	 */
+	public function hook() {
+
+		add_action( 'tec_tickets_commerce_flag_action_generated_attendee', [ $this, 'attendee_created' ], 10, 5 );
+		add_action( 'tec_tickets_commerce_ticket_deleted', tribe_callback( 'tickets.promoter.observer', 'notify_event_id' ), 10, 2 );
+		add_action( 'event_tickets_checkin', [ $this, 'checkin' ], 10, 2 );
+	}
+
+	/**
+	 * Action fired when a TC attendee is created
+	 *
+	 * @since TBD
+	 *
+	 * @param \WP_Post $attendee Attendee object
+	 */
```

Comment:
```suggestion
	/**
	 * Action fired when a TC attendee is created.
	 *
	 * @since TBD
	 *
	 * @param \WP_Post $attendee Attendee object.
	 */
```